app/controllers/reservations_controller.rb

```diff
@@ -19,15 +26,36 @@ def create
     end
   end
 
+  def update
+    if @reservation.update(reservation_params)
+      redirect_to reservations_path, notice: 'Reservation status was successfully updated.'
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @reservation.destroy
+    redirect_to reservations_url, notice: 'Reservation was successfully deleted.'
+  end
+
   private
 
   def set_screening
     @screening = Screening.find(params[:screening_id])
   end
 
+  def set_reservation
+    @reservation = Reservation.find(params[:id])
```

Comment:
Make sure this reservation includes every associated model used in the apropriate views